library/core/src/ptr/alignment.rs

```diff
@@ -370,3 +376,74 @@ enum AlignmentEnum {
     _Align1Shl62 = 1 << 62,
     _Align1Shl63 = 1 << 63,
 }
+
+#[cfg(kani)]
+#[unstable(feature="kani", issue="none")]
+mod verify {
+    use super::*;
+
+    // pub const fn of<T>() -> Self
+    #[kani::proof]
+    pub fn check_of_i32() {
+        let alignment = Alignment::of::<i32>();
+        assert!(alignment.as_usize().is_power_of_two());
+    }
+
+    // pub const fn new(align: usize) -> Option<Self>
+    #[kani::proof_for_contract(Alignment::new)]
+    pub fn check_new() {
+        let a = kani::any::<usize>();
+        let alignment_opt = Alignment::new(a);
+        match alignment_opt {
+            Some(alignment) => assert_eq!(alignment.as_usize(), a),
+            None => assert!(!a.is_power_of_two())
+        }
+    }
+
+    // pub const unsafe fn new_unchecked(align: usize) -> Self
+    #[kani::proof_for_contract(Alignment::new_unchecked)]
+    pub fn check_new_unchecked() {
+        let a = kani::any::<usize>();
+
+        unsafe {
+            let alignment = Alignment::new_unchecked(a);
+            assert!(alignment.as_usize() > 0);
+        }
+    }
+
+    // pub const fn as_usize(self) -> usize
+    #[kani::proof_for_contract(Alignment::as_usize)]
+    pub fn check_as_usize() {
+        let a = kani::any::<usize>();
+        if let Some(alignment) = Alignment::new(a) {
```

Comment:
Can you derive `kani::Arbitrary` instead?

Reply:
Asking as someone who still feels like being a Rust newbie: can you please be more precise as to how I'd do this?

Reply:
The easiest way is to use the `derive` macro together with the `safety_invariant` helper that we recently added. https://model-checking.github.io/kani/tutorial-nondeterministic-variables.html

Reply:
I tried to do this, and got:
```
error: implementation has missing stability attribute
```
Any ideas what needs to be fixed here?

Reply:
Also, where can I find information about this `safety_invariant` helper?

Reply:
The `safety_invariant` documentation is here: https://model-checking.github.io/kani/crates/doc/kani/derive.Arbitrary.html#type-safety-specification-with-the-safety_constraint-attribute

Reply:
> I tried to do this, and got:
> 
> ```
> error: implementation has missing stability attribute
> ```
> 
> Any ideas what needs to be fixed here?

Yes, every method in the standard library needs a stability attribute. We've been adding an `unstable` attribute with "kani" as the reason